### PR TITLE
Admin Dashboard Delete Confirmation Dialog

### LINKED ISSUE #21: Add Confirmation on DB Delete
- [ ] Moderators should be prompted to confirm they want to delete a submission

### PR DESCRIPTION
**Pull request by theyokohamalife**
## Resolves #21

I created a smaller PR for the admin panel, because the other one started to get too big and complicated. This one is just focused on breaking the admin panel down into smaller components. Styling changes will come later. Also I couldn't abstract the `v-dialog` for editing, so would really appreciate if someone else could tackle that one!

What this PR does:

1. Creates a `PendingSubmissions` component
2. Creates a `DeleteDialog` component
3. Abstracts the table headers to `dbTableHeaders.json`

### How to test

Create a fake clinic submission on the site preview and delete it.

### Screenshots
Screen Shot 2021-06-20 at 2 56 30 PM (2)

### CODE REVIEW

**Comment by theyokohamalife** [comment]
> Great stuff.
> 
> Re:
> 
> > Also I couldn't abstract the v-dialog for editing, so would really appreciate if someone else could tackle that one!
> 
> Was there a specific change you wanted to add to it?

Just to make it a separate component so it can be used to edit pending submissions and existing data in the db.

File: components/PendingSubmissions.vue [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        this.pendingItem[key] = item[key];
+      }
+      this.editedItem = Object.assign(item);
+    },
+    async deleteItem(item) {
+      const itemIndex = this.clinics.indexOf(item);
+      try {
+        await this.$fireModule
+          .firestore()
+          .collection("pending")
```
[the remark this answers is not in the document]
Comment (theyokohamalife) [reply components/PendingSubmissions.vue:141]: Hmm any suggestions on what I should name it and where I should move it to?

[the remark this answers is not in the document]
Comment (renderf0x) [reply components/PendingSubmissions.vue:140]: In an ideal world, there would be a DbManager class, and this view would have no knowledge of firebase or specific implementations, then you could just call a class method. Not really needed yet though =).

In the near term, we could define a DbConstants file that could just contain these, so you can reference collection names by `DbConstants.collections.PENDING`, or similar.

Again, super nit, so feel free to ignore for now!

File: data/adminDbTableHeaders.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (renderf0x) [inline data/adminDbTableHeaders.json:1]: ty!

**Comment by renderf0x** [comment]
> > Great stuff.
> > Re:
> > > Also I couldn't abstract the v-dialog for editing, so would really appreciate if someone else could tackle that one!
> > 
> > 
> > Was there a specific change you wanted to add to it?
> 
> Just to make it a separate component so it can be used to edit pending submissions and existing data in the db.

Ah I see. Creating a generic dialog component, ne. If we want to preserve inline styling that could make sense; otherwise I'd say `<v-dialog />` is itself the abstraction.

File: nuxt.config.js [hunk 3]
```diff
@@ -101,7 +101,7 @@ export default {
       {
         config: {
           /* eslint-disable-next-line no-undef */
-          apiKey: process.env.FIREBASE_API_KEY,
+          apiKey: process.env.LASHAWN_FIREBASE_API_KEY,
```
Comment (renderf0x) [inline nuxt.config.js:104]: Did you specifically want to have your name in the public repo here? (np if so, just wanted to check)

**Review by renderf0x — APPROVED** [review]
LGTM overall!

File: nuxt.config.js (hunk 3 above)
Comment (theyokohamalife) [reply nuxt.config.js:104]: Nope! I didn't mean to check that in. Thank you for catching it!

**Comment by Ermish** [comment]
LGTM!

**Pull request merged**